### ISSUE TITLE
Add button to share in Mastodon

### ISSUE THREAD

**Issue by victorhck** [openSUSE/jekyll-theme#7]
Hello!
I would love to have a button in the footer to share the news articles in Mastodon. The [fsfe.org](https://fsfe.org/about/about.es.html) has a service for that, but don't know where find the code and how to implement...
In my blog I also use a [service](https://gitlab.com/victorhck/share_freedom) to share in free social services, like Mastodon, but don't know if it's possible to adapt...
Greetings.

**Comment by guoyunhe** [comment]
It would be better to have a share link, without any JavaScript. Example https://simplesharebuttons.com/html-share-buttons/

JS sharing buttons are constantly violating user privacy. So it is better to not add any JS related to sharing buttons.

**Comment by vinzv** [comment]
Yes, but we need something that offers the host selection for Mastodon hosts.

**Comment by guoyunhe** [comment]
As long as we get all site links, we can do it without additional JavaScript.

**Comment by hellcp** [comment]
Oh how naive, it is impossible to do without js if you want to do this with a static site

**Comment by guoyunhe** [comment]
it is possible with some tricks of checkbox+label https://stackoverflow.com/a/48431586/4699409

**Comment by vinzv** [comment]
> As long as we get all site links, we can do it without additional JavaScript.

That's impossible. There are literally thousands and the number is constantly changing. So have an input field is the only future-proof way of doing this. Plus this doesn't need JS.

**Comment by hellcp** [comment]
It does need js, you can't post a form to a custom url without js

**Comment by vinzv** [comment]
Ah, right. Forgot about that. So we're basically back to this one:
https://aly-ve.github.io/Mastodon-share-button/

> JS sharing buttons are constantly violating user privacy. So it is better to not add any JS related to sharing buttons.

If they are hosted somewhere else. Which we don't want to.

**Comment by victorhck** [comment]
Don't know if can be useful. In the FSFE web there is a button to shere in "fediverse" (where you add the server you want to share. Ex: mastodon.social)

https://fsfe.org/news/2020/news-20200506-01.es.html
